### ISSUE TITLE
mdButton adds aria-label when element already has text content

### ISSUE THREAD

**Issue by marcysutton** [angular/material#8789]
**Actual behavior**:
-  What is the issue? *

mdButton is erroneously adding an `aria-label` attribute when it shouldn't because there's already text in the element. Text in a button element (custom or native, doesn't matter) is announced just fine on focus. 

-  What is the expected behavior?

An `aria-label` is not added when the button already has text, and it doesn't warn for a missing attribute.

The problem is in $mdAria, either in the `expectWithText` method or `expect` method called internally. The `expectWithText` method seems more appropriate for `md-radio-button`, which DOES need an `aria-label` even when there is child text content (required to expose an accessible name). The button expect version needs the same interpolation support, though.

**CodePen** or Steps to reproduce the issue: *
Visible on the button demos on https://material.angularjs.org
Inspect the buttons with text content, observe the extra `aria-label`

**Angular Versions**: *
Current

**Additional Information**:
-  Browser Type: * 
All
-  Browser Version: *
All
-  OS: *
All

----
Shortcut to create a [new CodePen Demo](http://codepen.io/team/AngularMaterial/pen/bEGJdd).
Note: `*` indicates required information. Without this information, your issue may be auto-closed.

> Do not modify the titles or questions. Simply add your responses to the ends of the questions.
  Add more lines if needed
